src/state/lists/updater.ts

```diff
@@ -23,7 +23,8 @@ export default function Updater(): null {
   const listsState = useAppSelector((state) => state.lists)
 
   useEffect(() => {
-    TokenSafetyLookupTable.update(listsState)
+    const keysArePresent = [UNI_EXTENDED_LIST, UNI_LIST, ...UNSUPPORTED_LIST_URLS].every((key) => listsState.byUrl[key])
```

Comment:
This is essentially the same as checking for hydration, but doesn't check for type safety at the point of use. Can you amend it in two ways:

- check for type safety at the point of use (ie in `tokenSafetyLookup#update`)
- this check is essentially the same as checking for hydration, but it couples `keysArePresent` to the implementation of `tokenSafetyLookup`. Can you instead add a hook to check if the state is hydrated? (if you search for "hydrated" you'll find similar logic already in the codebase.)

Reply:
1. is proving difficult while retaining the flexibility of string indexing byUrl